src/main/cpp/SwerveDrive.cpp

```diff
@@ -147,24 +150,44 @@ void SwerveDrive::drive(units::meters_per_second_t dx, units::meters_per_second_
         -GeneralConstants::MAX_VOLTAGE, GeneralConstants::MAX_VOLTAGE) );
 
     flModule_.setSpeedMotor( 0.2*std::clamp(fl_opt.speed.value(), -1.0, 1.0) );
+    // frc::SmartDashboard::PutNumber("fl speed error", fl_opt.speed.value() - flModule_.getVelocityMPS().value());

```

Comment:
what is the opt thing here (fl_opt, fr_opt, etc) and what is the yaw thing on line 135(navx_yaw) and what do these two things do?